### PR TITLE
#7 security basics

### PR DESCRIPTION
**Pull request by yanna-stepanova**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -115,6 +115,12 @@
 			<artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
 			<version>2.5.0</version>
 		</dependency>
+		<dependency>
+			<groupId>org.apache.struts</groupId>
```
Comment (Elena-Bruyako) [inline pom.xml]: why do you need this dependency?

File: src/main/java/com/yanna/stepanova/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@RequiredArgsConstructor
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepo;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepo.findByEmail(requestDto.email()).isPresent()) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/yanna/stepanova/service/impl/UserServiceImpl.java]: It's better to create existsByEmail() method and avoid calling isPresent

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

File: pom.xml (hunk 1 above)
Comment (yanna-stepanova) [reply pom.xml]: This dependency was added for using @FieldMatch in UserRegistrationRequestDto and not writing custom annotation. But validating two fields (password & repeatPassword) for a match still doesn't work

File: src/main/java/com/yanna/stepanova/dto/user/UserRegistrationRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.yanna.stepanova.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import org.apache.struts.beanvalidation.constraints.FieldMatch;
+import org.hibernate.validator.constraints.Length;
+
+@FieldMatch(first = "password", second = "repeatPassword", message = "These passwords must match")
```
Comment (MAMentorFX) [inline src/main/java/com/yanna/stepanova/dto/user/UserRegistrationRequestDto.java]: It's possible that @FieldMatch doesnt work with records.

Also According to https://www.baeldung.com/spring-mvc-custom-validator
consider trying @FieldsValueMatch(
      field = "password", 
      fieldMatch = "repeatPassword", 
      message = "Passwords do not match!"
    ),

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

**Review by koval-kn — APPROVED** [review]

**Pull request merged**